### DISCUSSION TITLE
Add "legacy" keybinds for old push-to-talk behavior

### DISCUSSION THREAD

**Discussion by zkxs** [FakeFishGames/Barotrauma#9373] [Ideas and suggestions]
Please add the ability for players to opt in to the old push-to-talk behavior.

## Background
**The old system:** separate keys for radio and local push-to talk.  
**The new system:** one push-to-talk key, and a key to cycle chat mode between radio/local

For text chat, there are legacy binds available for people who want the old behavior. For voice chat, there is no option for players to revert to the old behavior. For consistency, I'd expect either both or neither chat systems to allow opting-in to old behavior.

## Justification
I've seen the argument made that the old push-to-talk-system was "over clunky", and we just need time to adjust. To that, I say how so? Both systems require two keys, while the new system also requires us to keep track of a boolean. This seems like objectively more mental overhead to me.

## My Personal Experience
I play plenty of games that have multiple voice channels. (Planetside 2 for example has *seven*!) Every other game I've played just has a separate key for each voice channel, and that's how I prefer it. I find it easier to just commit to muscle memory "this type of talking maps to this finger position" than keeping track of some internal game state that governs where my voice will go.

**Comment by Foxlo** [comment]
100% this. 

The new system is double the key press for voice activity players and encourages bad radio etiquette (short comms, clear communications etc.)

There is no reason to force this on the players. It's not QoL, it's not something that was damaging the gameplay in any shape or form. It's purely and entirely a preference, and the new default makes less sense.

From the account of one of the unstable testers, they didn't test the VOIP because they played on text chat. Well as someone who exclusively plays VOIP and only plays with VOIP people, me and all our server dislike it.

**Comment by Jade-Harleyy** [comment]
i agree 💯

**Comment by IConnC** [comment]
Agreed, my community uses radio's heavily and it becomes very convoluted and difficult to keep radio chatter clean with this update.

**Comment by ekqvist** [comment]
Hello, everyone!

Thanks for your feedback. I understand where you're coming from and understand the want to retain the old voice chat behavior with separate binds. It may have been a slight oversight to remove the separate voice chat binds seeing how we retained the old text chat binds as "legacy" keybinds.

Someone also posted a bug report about this (#9379) and I have now repurposed that as a feature request to reintroduce the separate binds as an option for people who still want to continue using them. We will hopefully look into this issue when we can, but it might take some time for the change to reach the public release.

Sorry for the inconvenience!
